src/scripts/Update.ts

```diff
@@ -1129,6 +1129,16 @@ class Update implements Saveable {
             if (saveData.party.caughtPokemon.filter(p => p.id === 103.02).length) {
                 saveData.wallet.currencies[1] += 50000;
             }
+
+            // Replace Pokémon names to IDs
+            saveData.breeding.eggList?.forEach(pokemonName => {
+                const pokemonEgg = PokemonHelper.getPokemonByName(pokemonName);
+                Update.renamePokemonInSaveData(saveData, pokemonName, pokemonEgg.id);
```

Comment:
It's maybe a bit hack to use this function here. I was confused when i read it.
I am not totally sure how i think it should look, but if people uses `renamePokemonInSaveData` later on, it will break the save. So we either need to rename the function, or remove it. But if we just rename it, the old update functions will look weird...
If you can't find a good solution either, i will try to find something, when i get the PR back

Reply:
From what I understand, `renamePokemonInSaveData` is only used to change the Pokémon name in the hatchery and the queue, so with these changes, it wouldn't be needed anymore (unless you want to change the Pokémon ID).
Would adding `renamePokemonInSaveData` later on actually break the save? Let's say I want to change the Nidoran (F) name using the rename function, the function wouldn't find `Nidoran (F)` in the hatchery because it's saved as `29` and just do nothing.
It works like this, right?

Reply:
Yeah, you are right. Should not break anything. But the function will then be sorta useless in later update functions, so we can maybe rename it to `changeHatcheryKey` or something?